lib/elliptic/eddsa/index.js

```diff
@@ -0,0 +1,185 @@
+var elliptic = require('../../elliptic');
+var utils = elliptic.utils;
+var assert = utils.assert;
+var KeyPair = require('./key');
+
+function parseBytes(bytes) {
+  return typeof bytes === 'string' ? utils.toArray(bytes, 'hex') :
+                                     bytes;
+}
+
+function EDDSA(curve) {
+  if (!(this instanceof EDDSA))
+    return new EDDSA(curve);
+
+  assert(typeof curve === 'string');
+  assert(curve === 'ed25519', 'only tested with ed25519 so far');
+  var curve = this.curve = elliptic.curves[curve].curve;
+  this.g = curve.g;
+  this.g.precompute(curve.n.bitLength() + 1);
+
+  this.pointClass = this.g.mul(new bn(2)).constructor;
+  this.encBytes = Math.ceil(curve.n.bitLength() / 8);
+  this.hash = hash.sha512;
+}
+
+module.exports = EDDSA;
+
+/**
+* @param {Array} message - message bytes
+* @param {Array|String|KeyPair} secret - secret bytes or a keypair
+* @returns {bytes} - signature
+*/
+EDDSA.prototype.sign = function sign(message, secret) {
+  message = parseBytes(message);
+  var key = this.keyFromSecret(secret);
+  var r = this.hashToInt(key.prefix, message);
+  var R = this.encodePoint(this.g.mul(r.mod(this.curve.n)));
+  var s_ = this.hashToInt(R, key.pubBytes, message).mul(key.priv);
+  var S = r.add(s_).mod(this.curve.n);
+  return R.concat(this.encodeInt(S));
+};
+
+function pointEq(a, b) {
+  return a.getX().cmp(b.getX()) === 0 &&
+         a.getY().cmp(b.getY()) === 0;
+}
+
+EDDSA.prototype.isPoint = function isPoint(val) {
+  var isPoint = val instanceof this.pointClass;
+  return isPoint;
+};
+
+/**
+* @param {Array} sig - sig bytes
+* @param {Array} message - message bytes
+* @param {Array|String|Point|KeyPair} pub - public key
+* @returns {bytes} - true if public key matches sig of message
+*/
+EDDSA.prototype.verify = function verify(message, sig, pub) {
+  sig = parseBytes(sig);
+  message = parseBytes(message);
+  var R = sig.slice(0, this.encBytes);
+  var key = this.keyFromPublic(pub);
+  var S = this.decodeRedInt(sig.slice(this.encBytes));
+  var h = this.hashToInt(R, key.pubBytes, message).mod(this.curve.n);
+  var SG = this.g.mul(S);
+  var RplusAh = this.decodePoint(R).add(key.pub.mul(h));
+  return pointEq(RplusAh, SG);
+};
+
+/**
+* @returns {bn} little-endian intepretation of hash of one or more
+*               Arrays of bytes.
+*/
+EDDSA.prototype.hashToInt = function hashToInt() {
+  var hash = this.hash();
+  for (var i = 0; i < arguments.length; i++)
+    hash.update(arguments[i]);
+  return this.decodeInt(hash.digest());
+};
+
+EDDSA.prototype.keyFromPublic = function keyFromPublic(pub) {
+  if (pub instanceof KeyPair) {
+    return pub;
+  } else {
+    var pubBytes;
+    if (this.isPoint(pub)) {
+      pubBytes = this.encodePoint(pub);
+    } else {
+      pubBytes = parseBytes(pub);
+      pub = this.decodePoint(pubBytes);
+    }
+    return new KeyPair(this, { pub: pub, pubBytes: pubBytes });
+  }
+};
+
+EDDSA.prototype.keyFromSecret = function keyFromSecret(secret) {
```

Comment:
I don't want to use private, perhaps `seed` would be better? because as you can see the secret/seed is hashed, and the first half is used, bar a little bit twiddling as the `priv`, the scalar multiplied by the base point to find the public key. The remaining half is used as a prefix to the message hash. 

`keyFromPrivate` would be more consistent with `EC` but ..... ? do you like secret or seed ? or you're OK with private?